ethereum/rpc/backend/backend.go

```diff
@@ -244,14 +244,14 @@ func (e *EVMBackend) HeaderByNumber(blockNum types.BlockNumber) (*ethtypes.Heade
 
 	req := &evmtypes.QueryBlockBloomRequest{}
 
-	res, err := e.queryClient.BlockBloom(types.ContextWithHeight(resBlock.Block.Height+1), req)
+	blockBloomResp, err := e.queryClient.BlockBloom(types.ContextWithHeight(resBlock.Block.Height), req)
```

Comment:
I think there are other BlockBloom queries that also return error

Reply:
yeah ok, I will also do the change.

I am not sure if I should remove the warning log if it is an expected behavior?

Reply:
bloom filter does not seems to be used in the filter

Reply:
ok it seems the bloom filter logic is just not working. I will try to fix this in this PR or another PR

Reply:
created an issue to track this https://github.com/tharsis/ethermint/issues/277

Reply:
BTW this PR is still usefull because it most case blockbloom will be empty (no events in contract)